studio/components/interfaces/TableGridEditor/SidePanelEditor/ForeignKeySelector/ForeignKeySelector.tsx

```diff
@@ -313,21 +315,34 @@ const ForeignKeySelector = ({
                   ))}
                 </Listbox>
               )}
-              {!matchingColumnTypes && (
+              {!matchingColumnTypes && !column?.format && (
+                <Alert_Shadcn_ variant="default">
+                  <IconAlertCircle className="h-4 w-4" />
+                  <AlertTitle_Shadcn_>The referenced column has no type</AlertTitle_Shadcn_>
```

Comment:
i think lets simplify the messaging here

title: The reference column's type will be updated to {selectedColumn.data_type}
description: The referenced column {column.name} must match the type of the selected foreign column when creating a foreign key relationship.